### ISSUE TITLE
Type hinting

### ISSUE THREAD

**Issue by RobinD42** [wxWidgets/Phoenix#304]
Add type hinting to the `.pyi` stub files, as well as `@overload` decorators to document the C++ overloaded methods. See https://www.python.org/dev/peps/pep-0484/

**Comment by lvu** [comment]
+1 for this; would be really helpful!

**Comment by USSX-Hares** [comment]
Really waiting for this feature

**Comment by USSX-Hares** [comment]
Type hinting could be automated, isn't it?
You're already generating docs in the automated mode

**Comment by RobinD42** [comment]
Yes, it could be automated, it just needs updates to `etgtools/pi_generator.py` to make it happen. If you're interested in doing it please do and submit a PR.
